src/ConsumerConfig.h

```diff
@@ -21,14 +21,17 @@
 #define CONSUMER_CONFIG_H
 
 #include <pulsar/c/consumer_configuration.h>
+#include "ThreadSafeDeferred.h"
 #include "MessageListener.h"
 
 #define MIN_ACK_TIMEOUT_MILLIS 10000
 
 class ConsumerConfig {
  public:
-  ConsumerConfig(const Napi::Object &consumerConfig, pulsar_message_listener messageListener);
+  ConsumerConfig();
   ~ConsumerConfig();
+  void InitConfig(const std::shared_ptr<ThreadSafeDeferred> deferred, const Napi::Object &consumerConfig,
```

Comment:
Don't pass const value as the parameter type. For example,

```c++
struct Foo {
  void f1() const {}
  void f2() {}
};

void f(const std::shared_ptr<Foo> foo) {
  foo->f1();
  foo->f2();  // OK, even though f2() is a non-const method
}
```

`f()` can only guarantee `foo` itself cannot be modified like `foo.set(nullptr)`.

However, since this method only needs a reference to the `ThreadSafeDeferred` object and does not transfer or share the ownership of `deferred` to another thread, I suggest passing a reference here.

```c++
  void InitConfig(ThreadSafeDeferred &deferred, const Napi::Object &consumerConfig,
                  pulsar_message_listener messageListener);
```

Then call `deferred.Reject` instead of `deferred->Reject` in the implementation.

BTW, I still don't get why this refactoring is needed in this PR.

Reply:
Got it. Thanks for your reminder.

Reply:
I will keep use std::shared_ptr<ThreadSafeDeferred>, because it will create by NAPI method: `auto deferred = ThreadSafeDeferred::New(info.Env());`